code/game/machinery/contraband_detector.dm

```diff
@@ -0,0 +1,215 @@
+	if(!wires.is_cut(WIRE_CONTRADETECT_IDENTIFIER) && (((identifier_wire_pulsed_until > world.time) && identifier_wire_pulsed_until != 0) || (istype(A) && A.has_contraband())))
+		if(ismob(A))
+			if(mob_cooldown < world.time)
+				mob_cooldown = world.time + 1 SECOND
+			else
+				return
+		else
+			if(obj_cooldown < world.time)
+				obj_cooldown = world.time + 2 SECONDS
+			else
+				return
+		flick("detected", src)
+		visible_message(SPAN_WARNING("\The [src] detects contraband on \the [A.name]!"))
+		if(!wires.is_cut(WIRE_CONTRADETECT_ALARM))
+			var/area/AR = get_area(src)
+			announce.autosay("Contraband detected on \the [A] at \the [AR.name]!", "Contraband Detection System")
+
+/obj/machinery/contraband_detector/proc/announce_tampering()
+	var/area/AR = get_area(src)
+	announce.autosay("ALERT! Tampering detected on \the [src] at [AR.name]!", "Contraband Detection System")
+
+/obj/machinery/contraband_detector/powered()
+	if(wires.is_cut(WIRE_MAIN_POWER1) || wires.is_cut(WIRE_MAIN_POWER2))
+		return FALSE
+	return ..()
+
+/obj/machinery/contraband_detector/on_update_icon()
+	if(stat & BROKEN)
+		icon_state = "broken"
+	else if(stat & NOPOWER)
+		icon_state = "unpowered"
+	else
+		icon_state = "powered"
+
+/obj/machinery/contraband_detector/attackby(obj/item/I, mob/user)
+	if(isScrewdriver(I))
+		if(id_locked && !panel_open)	// you can close it when ID locked, just not open it
+			to_chat(user, SPAN_NOTICE("\The [src]'s panel is locked!"))
+		else
+			panel_open = !panel_open
+			user.visible_message(SPAN_WARNING("[user] screws the camera's panel [panel_open ? "open" : "closed"]!"),
+			SPAN_NOTICE("You screw the camera's panel [panel_open ? "open" : "closed"]."))
+			playsound(src.loc, 'sound/items/Screwdriver.ogg', 50, 1)
+	if((isWirecutter(I) || isMultitool(I)) && panel_open)
+		return wires.Interact(user)
+	if(isWrench(I))
+		if(wrenching)
+			to_chat(user, SPAN_WARNING("Someone is already [anchored ? "un" : ""]securing the detector!"))
+			return
+		if(!anchored && isinspace())
+			to_chat(user, SPAN_WARNING("Cannot secure detectors in space!"))
+			return
+
+		user.visible_message(SPAN_WARNING("[user] begins [anchored ? "un" : ""]securing the contraband detector."), \
+			SPAN_NOTICE("You begin [anchored ? "un" : ""]securing the contraband detector."))
+
+		wrenching = TRUE
+		if(do_after(user, 10 SECONDS, src))
+			if(!anchored)
+				playsound(loc, 'sound/items/Ratchet.ogg', 100, 1)
+				anchored = TRUE
+				update_icon()
+				to_chat(user, SPAN_NOTICE("You secure the exterior bolts on the contraband detector."))
+				RegisterSignal(loc, COMSIG_ENTERED, .proc/detect_contraband)
+			else if(anchored)
+				playsound(loc, 'sound/items/Ratchet.ogg', 100, 1)
+				anchored = FALSE
+				to_chat(user, SPAN_NOTICE("You unsecure the exterior bolts on the contraband detector."))
+				update_icon()
+				UnregisterSignal(loc, COMSIG_ENTERED)
+		wrenching = FALSE
```

Comment:
This is handled by obj/machinery

If you need to override for checks just do the check and then return ..(args)